ocaml/xapi/xapi_host.ml

```diff
@@ -2591,13 +2591,34 @@ let write_uefi_certificates_to_disk ~__context ~host:_ =
             let path = Filename.concat !Xapi_globs.varstore_dir name in
             Unixext.unlink_safe path
           )
-          ["KEK.auth"; "db.auth"] ;
+          ["PK.auth"; "KEK.auth"; "db.auth"; "dbx.auth"] ;
         (* No uefi certificates, nothing to do. *)
         if contents <> "" then (
           with_temp_file_contents ~contents
             (Tar_unix.Archive.extract extract_certificate_file) ;
           debug "UEFI tar file extracted to temporary directory"
-        )
+        ) ;
+        (* If some mandatory auth files are missing, fetch them from a fallback dir *)
+        List.iter
+          (fun name ->
+            let path = Filename.concat !Xapi_globs.varstore_dir name in
+            let fallback_path =
+              Filename.concat !Xapi_globs.fallback_auth_dir name
+            in
+            if (not (Sys.file_exists path)) && Sys.file_exists fallback_path
```

Comment:
Is this sufficient? What about the file at `path` is more recent than the file at `fallback_path`? Would you want to create a new fallback?

Reply:
This is not what the code do. (Or I misunderstood your comment)
In the case the extract of the XAPI db tarball has no file (file at `path` doesn't exist) the code fetch a fallback file and copy it to `path`.
To update the file in the tarball, an update of varstored (for exemple) install a new file at fallback path & call `Pool.set_uefi_certificate` with a tarball containing the new file.

Reply:
Sorry, I misread this. The code is fetching a file if it is missing, not creating the fallback as I had read it.

Reply:
Exactly, the XAPI does not change/delete the fallback file, only reads it.
To change it it'd come from another process.
IIUC your usecase `PK` would be modified by a new `varstored` RPM and `dbx` by a new `secureboot-certs`, the %post of the RPM would install the new auth files to the fallback directory and the XAPI db would need to be updated (to propagate to the whole pool) with the `Pool.set_uefi_certificates` method with a tarball with the updated auth files. :)